GCMAggregatingTableViewDataSource/GCMAggregatingTableViewDataSource/GCMAggregatingTableViewDataSource.m

```diff
@@ -57,12 +57,25 @@ - (void)addChildDataSource:(id<UITableViewDataSource>)childDataSource {
   [self.childDataSources addObject:childDataSource];
 }
 
+- (void)addChildDataSource:(id<UITableViewDataSource>)childDataSource afterDataSource:(id<UITableViewDataSource>)precedingDataSource{
+  if ([self.childDataSources containsObject:precedingDataSource]) {
+    NSInteger index = [self.childDataSources indexOfObject:precedingDataSource];
+    [self.childDataSources insertObject:childDataSource atIndex:index + 1];
+  }
```

Comment:
I'd suggest that if precedingdataSource can't be found, we should probably assert. Failing silently in this situation seems like it would cause confusion as to why the new datasource isn't appearing.